Source/Wikitivity.Agent/WikitivityAgentHelper.cs

```
@@ -5,43 +5,156 @@
+		{
+			QueryResultSlim test = new QueryResultSlim();
+			List<int> CasesWithWikitivity = new List<int>();
+
+			QueryRequest omGetInitialJobQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request Url' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, 
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
 				{
-					success = true;
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"}
+
 				}
+			};
+			try
+			{
+				test = await proxy.QuerySlimAsync(workspaceID, omGetInitialJobQuery, 1, 10000);
+			}
+			catch (Exception e)
+			{
+				Console.WriteLine(e);
+				throw;
+			}
+
+			foreach (var singleWorkspace in test.Objects)
+			{
+				CasesWithWikitivity.Add(singleWorkspace.ArtifactID);
+			}
+			return CasesWithWikitivity;
+		}
+		
+		public static async Task<QueryResultSlim> FindAllWikiCases(IObjectManager proxy, int workspaceID)
+		{
+
+
+			QueryRequest omGetInitialJobQuery = new QueryRequest()
+			{
+				ObjectType = new ObjectTypeRef() { Guid = WikitivityRDOGuid },
+				Condition = "('Request Url' ISSET)",
+				IncludeIDWindow = false,
+				RelationalField = null, //name of relational field to expand query results to related objects
+				SampleParameters = null,
+				SearchProviderCondition = null,
+				Sorts = null,
+				Fields = new List<FieldRef>()
+				{
+					new FieldRef() {Name = "Request Url"},
+					new FieldRef() {Name = "Request ID"},
+					new FieldRef() {Name = "Page Title"}
+				}
+			};
+
+			try
+			{
+				QueryResultSlim queryResults = await proxy.QuerySlimAsync(workspaceID, omGetInitialJobQuery, 1, 10000);
+				return queryResults;
 			}
 			catch (Exception ex)
 			{
-				throw new Exception("An error an occurred when deleting completed requests", ex);
+				return new QueryResultSlim();

```

We are eating the exception here which is bad. Instead in the try block, I would see if the results count is zero then I would return an empty list.

---

Any exception like this where you "handle it" by eating it and defaulting should be logged, even if just at debug level.  It's important to know this happens.